gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapper.java

```diff
@@ -0,0 +1,195 @@
+            .map(RCMRMT030101UK04Component2::getEhrSupplyPrescribe)
+            .map(supplyPrescribe -> mapToOrderMedicationRequest(medicationStatement, supplyPrescribe))
+            .filter(Objects::nonNull)
+            .map(medicationRequest -> setCommonFields(
+                medicationRequest, requester, recorder, subject, context, authoredOn
+            ))
+            .collect(Collectors.toList());
+
+        List<MedicationRequest> medicationRequestsPlan = medicationStatement.getComponent()
+            .stream()
+            .filter(RCMRMT030101UK04Component2::hasEhrSupplyAuthorise)
+            .map(RCMRMT030101UK04Component2::getEhrSupplyAuthorise)
+            .map(supplyAuthorise -> mapToPlanMedicationRequest(ehrExtract, medicationStatement, supplyAuthorise))
+            .filter(Objects::nonNull)
+            .map(medicationRequest -> setCommonFields(
+                medicationRequest, requester, recorder, subject, context, authoredOn
+            ))
+            .collect(Collectors.toList());
+
+        List<MedicationStatement> medicationStatements = medicationStatement.getComponent()
+            .stream()
+            .filter(RCMRMT030101UK04Component2::hasEhrSupplyAuthorise)
+            .map(RCMRMT030101UK04Component2::getEhrSupplyAuthorise)
+            .map(supplyAuthorise -> mapToMedicationStatement(medicationStatement, supplyAuthorise))
+            .filter(Objects::nonNull)
+            .peek(medicationStatement1 -> {
+                context.ifPresent(context1 -> medicationStatement1.setContext(new Reference(context1)));
+                medicationStatement1.setSubject(new Reference(subject));
+                medicationStatement1.setEffective(authoredOn);
+                medicationStatement1.setDateAssertedElement(authoredOn);
+            })
+            .collect(Collectors.toList());
+
+        return Stream.of(medications, medicationRequestsOrder, medicationRequestsPlan, medicationStatements)
+            .flatMap(List::stream)
+            .collect(Collectors.toList());
+    }
+
+    private MedicationStatement mapToMedicationStatement(RCMRMT030101UK04MedicationStatement medicationStatement,
+        RCMRMT030101UK04Authorise supplyAuthorise) {
+        return medicationStatementMapper.mapToMedicationStatement(medicationStatement, supplyAuthorise);
+    }
+
+    private MedicationRequest mapToOrderMedicationRequest(RCMRMT030101UK04MedicationStatement medicationStatement,
+        RCMRMT030101UK04Prescribe supplyPrescribe) {
+        return medicationRequestOrderMapper.mapToOrderMedicationRequest(medicationStatement, supplyPrescribe);
+    }
+
+    private MedicationRequest mapToPlanMedicationRequest(RCMRMT030101UK04EhrExtract ehrExtract,
+        RCMRMT030101UK04MedicationStatement medicationStatement, RCMRMT030101UK04Authorise supplyAuthorise) {
+        return medicationRequestPlanMapper.mapToPlanMedicationRequest(ehrExtract, medicationStatement, supplyAuthorise);
+    }
```

Comment:
I'm not sure if those methods are needed - they just simply pass the parameters without changing them, to methods with the same name

Reply:
removed these extra methods